import_data/tasks/tasks.py

```diff
@@ -819,26 +817,20 @@ def run_osm_update(ctx):
 
     change_file_path = f"{ctx.update_tiles_dir}/changes.osc.gz"
     lock_path = get_import_lock_path(ctx)
-    with FileLock(lock_path) as lock:
+    with FileLock(lock_path) as _lock:
         current_osm_timestamp = read_current_state(ctx)
         try:
             osmupdate_opts = _get_osmupdate_options(ctx)
-            ctx.run(
-                f'osmupdate {osmupdate_opts} {current_osm_timestamp} {change_file_path}'
-            )
+            ctx.run(f"osmupdate {osmupdate_opts} {current_osm_timestamp} {change_file_path}")
         except Failure as exc:
             if exc.result.return_code == 21:
-                logging.info('OSM state is up to date, no change to apply')
+                logging.info("OSM state is up to date, no change to apply")
                 return
             raise
 
         new_osm_timestamp = read_osm_timestamp(ctx, change_file_path)
 
-        osm_update(
-            ctx,
-            f"postgis://{ctx.pg.user}:{ctx.pg.password}@{ctx.pg.host}:{ctx.pg.port}/{ctx.pg.database}",
-            change_file_path
-        )
+        osm_update(ctx, _pg_conn_str(ctx), change_file_path)
```

Comment:
:boom: This is not equivalent.  
`_pg_conn_str` connects to `ctx.pg.import_database`, whereas the OSM update should be applied on `ctx.pg.database`

I would suggest to require the database name to be passed to `_pg_conn_str` to be as explicit as possible, as this could easily be overlooked.